app/models/concerns/status/safe_reblog_insert.rb

```diff
@@ -34,10 +36,15 @@ def _insert_record(values, returning)
       # Instead, we use a custom builder when a reblog is happening:
       im = _compile_reblog_insert(values)
 
-      connection.insert(im, "#{self} Create", primary_key || false, primary_key_value, returning: returning).tap do |result|
-        # Since we are using SELECT instead of VALUES, a non-error `nil` return is possible.
-        # For our purposes, it's equivalent to a foreign key constraint violation
-        raise ActiveRecord::InvalidForeignKey, "(reblog_of_id)=(#{values['reblog_of_id'].value}) is not present in table \"statuses\"" if result.nil?
+      with_connection do |_c|
+        connection.insert(
```

Comment:
I am not familiar with this code, but why are we taking `connection` as a method parameter, then calling `with_connection`, which seems to yield you a connection, and not use it?

Reply:
Why are we doing this? -- I believe this is just updated to reflect what changed in the actual rails class (including this non-use, see comment link).

Why is it like that? -- no idea.

Reply:
Good point, I missed that it was like that in the upstream file